### ISSUE TITLE
confusing simple search results

### ISSUE THREAD

**Issue by Antonialock** [pombase/website#873]
I wanted to see if we had a pombe ortholog to B3GLCT (we dont)

the "matches" in the simple search look really confusing, what do they match to?

<img width="416" alt="screen shot 2018-07-25 at 23 09 07" src="https://user-images.githubusercontent.com/12727506/43230383-ed4bf2c2-905f-11e8-80c7-d5f1c7e8e20e.png">

**Comment by ValWood** [comment]
Yes, I noticed the same when searching on human gene names yesterday. I think we should only find "exact matches" not fuzzy (if this is the problem)

**Comment by kimrutherford** [comment]
In this case I think it's tripping over the "3" in "B3GLCT".  It looks like it's treating digits as whitespace so it's searching for "B GLCT" and finding nonsense.  I'm sure that's fixable.

Another issue at the moment is that it searches the synonyms but only shows the term name.  If the synonym matches what you type that's what it should show, like in the query builder.

It would also be good if the matching part of the term name or synonym was highlighted.  Like if you search for "signal" and it finds "intracellular signal transduction" it should display it as:

 >  intracellular **signal** transduction

**Comment by kimrutherford** [comment]
> It looks like it's treating digits as whitespace so it's searching for "B GLCT" and finding nonsense. I'm
> sure that's fixable

I've fixed that locally but it's still not working very well.  There is no exact match for "B3GLCT" so it shows the closest matches.  Which aren't useful at all.  Let's chat about this.

**Comment by ValWood** [comment]
- [ ] Don't do fuzzy searching, but do partial match searching.

**Comment by kimrutherford** [comment]
I've fixed the search to only find exact matches or exact prefix matches.

It currently shows terms and publications that match any of the words you type.  Does that make sense?

**Comment by ValWood** [comment]
Hmm, I can't think of a way founf that. But who would search for "and" anyway?
Is it an issue??? I don't know...

**Comment by kimrutherford** [comment]
> Is it an issue??? I don't know...

OK.  Maybe give it a go as is and see if it's annoying?

**Comment by ValWood** [comment]
yep
